### PR TITLE
Move ingest-geoip default databases out of config

### PR DESCRIPTION
**Pull request by jasontedor**
Today the default databases bundled with ingest-geoip are treated as configuration files that we unbundle into the Elasticsearch configuration directory. This can cause problems for users using our Docker images if they bind mount over the configuration directory. Additionally, it creates complexity when trying to convert ingest-geoip to a module. This commit moves these databases out of the configuration directory and instead loads from the plugin directory. Further, custom databases can still be put into the configuration directory.

Relates #36898

### CODE REVIEW

**Comment by jasontedor** [comment]
@elasticmachine run the default distro tests

File: plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java [hunk 1]
```diff
@@ -66,48 +69,89 @@
...
+                    if (loadDatabaseOnHeap) {
+                        builder.fileMode(Reader.FileMode.MEMORY);
+                    } else {
+                        builder.fileMode(Reader.FileMode.MEMORY_MAPPED);
+                    }
+                    return builder.build();
+                });
+    }
+
+    private static void assertDatabaseExistence(final Path path, final boolean exists) throws IOException {
```
Comment (jakelandis) [inline plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java]: Should we move this to the unit tests to allow a user the ability to remove the default database(s) if they so choose.

Comment (jasontedor) [reply plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java]: Good question. I don't think we want them removing the default database files, as that means messing with the installation (poking around in the plugins directory for now, and the modules directory in the future). In fact, we have an explicit check that the default database files exist.

**Review by jakelandis — APPROVED** [review]
LGTM

**Review by rjernst — COMMENTED** [review]
One general question: could we move the default databases into resource files do they are loaded the same between tests and production? Then we don't need to mess with finding the directory they should exist in, but just load the resources as a stream.

**Comment by jasontedor** [comment]
No, because then we would not be able to memory map the databases as they have to be actual files on disk to do that (we can not memory map from a resource input stream).

**Pull request merged**